### PR TITLE
CON-2699 Blob SAS Token Generator

### PR DESCRIPTION
**Pull request by ChrisJWoodcock**

### CODE REVIEW

File: src/SFA.DAS.Assessor.Functions.ExternalApis/Config/OAuth.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿namespace SFA.DAS.Assessor.Functions.ExternalApis.Config
```
Comment (cofaulco) [inline src/SFA.DAS.Assessor.Functions.ExternalApis/Config/OAuth.cs:1]: this isn't all required now for use with Managed Identities

File: src/SFA.DAS.Assessor.Functions.Infrastructure/Options/PrintCertificates/BlobSasTokenGeneratorOptions.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+namespace SFA.DAS.Assessor.Functions.Infrastructure.Options.PrintCertificates
+{
+    public class BlobSasTokenGeneratorOptions
+    {
+        public int SasExpiryDays { get; set; }
+        public string SasIPAddress { get; set; }
+		public string SecureMessageTtl { get; set; }
```
Comment (cofaulco) [inline src/SFA.DAS.Assessor.Functions.Infrastructure/Options/PrintCertificates/BlobSasTokenGeneratorOptions.cs:7]: dodgy formatting

**Review by cofaulco — COMMENTED** [review]
A couple of minor comments added. 

Also:
File still with copied name: `When_command_executed_and_process_learner_returns_next_page - Copy.cs`

1 thing of note is that all the file renames and settings refactoring etc makes sense but will require a full regression test of the functions to ensure nothing has been adversely affected.

File: src/SFA.DAS.Assessor.Functions.ExternalApis/Config/OAuth.cs (hunk 1 above)
Comment (ChrisJWoodcock) [reply src/SFA.DAS.Assessor.Functions.ExternalApis/Config/OAuth.cs:1]: Removed

File: src/SFA.DAS.Assessor.Functions.Infrastructure/Options/PrintCertificates/BlobSasTokenGeneratorOptions.cs (hunk 2 above)
Comment (ChrisJWoodcock) [reply src/SFA.DAS.Assessor.Functions.Infrastructure/Options/PrintCertificates/BlobSasTokenGeneratorOptions.cs:7]: Done

File: src/SFA.DAS.Assessor.Functions/Domain/Print/Services/BlobFileTransferClient.cs [hunk 3]
```diff
@@ -224,11 +223,36 @@ private async Task<List<CloudBlob>> GetBlobsHierarchicalListingAsync(CloudBlobDi
...
+                SharedAccessExpiryTime = expiryTime,
+                Permissions = permissions ??
+                    SharedAccessBlobPermissions.Read |
+                    SharedAccessBlobPermissions.Write |
+                    SharedAccessBlobPermissions.Create |
+                    SharedAccessBlobPermissions.List |
+                    SharedAccessBlobPermissions.Delete
+            };
+
+            var ipAddressOrRange = !string.IsNullOrEmpty(ipAddress)
```
Comment (nbowes24) [inline src/SFA.DAS.Assessor.Functions/Domain/Print/Services/BlobFileTransferClient.cs:250]: Could this raise an error or warning if the IP Address is NullOrEmpty and not generate a token or send email

Comment (nbowes24) [reply src/SFA.DAS.Assessor.Functions/Domain/Print/Services/BlobFileTransferClient.cs:256]: Changes look good to me 👍

**Review by cofaulco — APPROVED** [review]

**Pull request merged**